[DRAFT/DO NOT MERGE]Testing creating a new page to navigate member and non-member to join Slack channel

siyunfeng opened the pull request:
<!--  Important! Add the number of the issue you worked on  --> 
This PR is to test how to create a new issue to solve 6129

### What changes did you make?
<!-- Note: add lines if needed, and remove any unused lines -->  
  - Created HTML file `join-slack-channel.html` to navigate how to join Slack channel
  - Changed the URL of **Join Slack Channel** from the corresponding slack channel to the `join-slack-channel.html`
  - Added an `onclick()` in the `<a>` element of **Join Slack Channel** to store the Slack channel URL in localStorage

### Why did you make the changes (we will use this info to test)?
<!-- Note: add lines if needed, and remove any unused lines -->  
  - Testing the potential solution to navigate non-members to join HfLA before joining the team Slack channel before writing instructions of creating the issue based on the ER 6129

<h3>CodeQL Alerts</h3>


After the PR has been submitted and the resulting GitHub actions/checks have been completed, developers should check the PR for CodeQL alert annotations.


<details><summary>Check the PR's comments. If present on your PR, the CodeQL alert looks similar as shown</summary>
  
![Screenshot 2024-10-28 154514](https://github.com/user-attachments/assets/ea66c586-c14c-45fd-8705-1c116224e704)


</details>

Please let us know that you have checked for CodeQL alerts. **Please do not dismiss alerts.**
- [x] I have checked this PR for CodeQL alerts and none were found.
- [ ] I found CodeQL alert(s), and (select one):
   - [ ] I have resolved the CodeQL alert(s) as noted
   - [ ] I believe the CodeQL alert(s) is a false positive (Merge Team will evaluate)
   - [ ] I have followed the Instructions below, but I am still stuck (Merge Team will evaluate)

<details><summary>Instructions for resolving CodeQL alerts</summary>

If CodeQL alert/annotations appear, refer to [How to Resolve CodeQL alerts](https://github.com/hackforla/website/issues/6463#issuecomment-2002573270).  

In general, CodeQL alerts should be resolved prior to PR reviews and merging

</details>

### Screenshots of Proposed Changes To The Website (if any, please do not include screenshots of code changes)
<!-- Notes: 
  - If there are no visual changes to the website, delete all of the script below and replace with "- No visual changes to the website"
  - If there are visual changes to the website, include the 'before' and 'after' screenshots below. 
  - If your images are too big, use the <img src="" width="" length="" />  syntax instead of ![image](link) to format the images
  - If images are not loading properly, you might need to double check the syntax or add a newline after the closing </summary> tag 
 --> 

<details>
<summary>Visuals before changes are applied</summary>

No related image

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
![join-slack-channel-page](https://github.com/user-attachments/assets/0e3f4229-bce7-4627-9455-72df124e8220)
hing_Files)

</details>

siyunfeng committed:
[DRAFT/DO NOT MERGE]Testing creating a new page to navigate member and non-member to join slack channel

daras-cu commented:
Hi @siyunfeng, I took a look at your code and I think the issues you are having are because `communities-of-practice.js` is imported as a JS module in `communities-of-practice.html`, therefore the functions within it are not accessible when called by the `onClick` event. Note the final line of the HTML file where the type is set to a module:
``` <script type="module" src="assets/js/communities-of-practice.js"></script> ```

To avoid changing how the script is handled, one possible solution is to use `communities-of-practice.js` to add an event listener to each `<a>` element that is a Slack link to handle the click event. Here is what I tried:
- In `communities-of-practice.html`, I added an additional class `slack-link` to the `<a>` element and also an id to hold the Slack URL
![html](https://github.com/user-attachments/assets/b671b44e-2db9-4a54-9321-905735e69315)

- Then in `communities-of-practice.js`, I modified the `setSlackLink` function slightly to get the Slack URL from the id attribute, and added an event listener to all the elements with the `slack-link` class in order to handle the click event
![js](https://github.com/user-attachments/assets/18baf9fe-aedb-459e-9374-512fd31c594d)

This worked to store and retrieve the URL in localStorage and print to console. There is still an issue retrieving the URL in `join-slack-channel.html` because you are trying to access local storage in a Liquid statement, which is resolved on the server side. You will likely need to add some more JS code or pass the URL in a different way, but hopefully this gets you started in the right direction. Let me know if everything makes sense and if you need additional help!

siyunfeng commented:
> Hi @siyunfeng, I took a look at your code and I think the issues you are having are because `communities-of-practice.js` is imported as a JS module in `communities-of-practice.html`, therefore the functions within it are not accessible when called by the `onClick` event. Note the final line of the HTML file where the type is set to a module: `<script type="module" src="assets/js/communities-of-practice.js"></script>`
> 
> To avoid changing how the script is handled, one possible solution is to use `communities-of-practice.js` to add an event listener to each `<a>` element that is a Slack link to handle the click event. Here is what I tried:
> 
> * In `communities-of-practice.html`, I added an additional class `slack-link` to the `<a>` element and also an id to hold the Slack URL
>   ![html](https://private-user-images.githubusercontent.com/104811751/421052194-b671b44e-2db9-4a54-9321-905735e69315.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2NDQxMzUsIm5iZiI6MTc0MTY0MzgzNSwicGF0aCI6Ii8xMDQ4MTE3NTEvNDIxMDUyMTk0LWI2NzFiNDRlLTJkYjktNGE1NC05MzIxLTkwNTczNWU2OTMxNS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxMFQyMTU3MTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNzljY2M5ZDcyZjQyM2NkMzJhOTFlMTZkYzQ2ZmM4NDU0OTliMjVkOGVkNGVkMTE0ZDFjZThiMmM2NTNiYTBhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ykr8jFL5VHVDiUg7snQQu7haZ4Xv7uSKWpiMODQoKcQ)
> * Then in `communities-of-practice.js`, I modified the `setSlackLink` function slightly to get the Slack URL from the id attribute, and added an event listener to all the elements with the `slack-link` class in order to handle the click event
>   ![js](https://private-user-images.githubusercontent.com/104811751/421056205-18baf9fe-aedb-459e-9374-512fd31c594d.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2NDQxMzUsIm5iZiI6MTc0MTY0MzgzNSwicGF0aCI6Ii8xMDQ4MTE3NTEvNDIxMDU2MjA1LTE4YmFmOWZlLWFlZGItNDU5ZS05Mzc0LTUxMmZkMzFjNTk0ZC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxMFQyMTU3MTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02Y2Q1M2E5YmZmNTI1Mzk2ZTRkMmM1NDU2NDdmZTJhNTQ2MzdkZjZmZDhkODJhZTg2ZjQ1NTQ2YmI4N2FhOWFkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.CjXARZ97de1Hg-JRHDIgRaW_zHEEtzy1ujzf-Hxu0cE)
> 
> This worked to store and retrieve the URL in localStorage and print to console. There is still an issue retrieving the URL in `join-slack-channel.html` because you are trying to access local storage in a Liquid statement, which is resolved on the server side. You will likely need to add some more JS code or pass the URL in a different way, but hopefully this gets you started in the right direction. Let me know if everything makes sense and if you need additional help!

These are helpful. Thank you @daras-cu !
I was able to get the value from the localStorage based on the code you posted. I'll start working on the instructions.
